datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -3082,180 +3087,6 @@ fn from_substrait_literal(
     Ok(scalar_value)
 }
 
-fn from_substrait_null(
-    consumer: &impl SubstraitConsumer,
-    null_type: &Type,
-    dfs_names: &[String],
-    name_idx: &mut usize,
-) -> Result<ScalarValue> {
```

Comment:
The match block below is effectively replaced by converting the Substrait Type to a DataFusion DataType, and then invoking `ScalarValue::try_from` which does the same thing effectively: https://github.com/apache/datafusion/blob/9eca7d165c3ddcd3449e833df9391b8216e0f5bc/datafusion/common/src/scalar/mod.rs#L3408-L3523